### PR TITLE
Implement non-blocking to_parquet

### PR DESCRIPTION
**Pull request by RobbeSneyders**
This is necessary to prevent OOM when processing large datasets. Added all the info in comments in the code.

### CODE REVIEW

File: src/fondant/component/data_io.py [hunk 1]
```diff
@@ -204,7 +207,34 @@ def _write_dataframe(self, dataframe: dd.DataFrame) -> dd.core.Scalar:
...
+        # https://dask.discourse.group/t/improving-pipeline-resilience-when-using-to-parquet-and-preemptible-workers/2141
+        to_parquet_tasks = [
+            d.to_parquet(
+                os.path.join(location, f"part.{i}.parquet"),
+                schema=pa.schema(list(schema.items())),
+            )
+            for (i, d) in enumerate(dataframe.to_delayed())
+        ]
+
+        client: dask.distributed.Client = dask.distributed.get_client()
```
Comment (mrchtr) [inline src/fondant/component/data_io.py:231]: `get_client()` will retrieve the global client I guess. But shouldn't we use the client we have initialised in the setup method explicitly to make clear where the client comes from?

**Review by mrchtr — APPROVED** [review]
Thanks @RobbeSneyders. It seems like we have finally a robust solution for our oom issues.

File: src/fondant/component/data_io.py (hunk 1 above)
Comment (RobbeSneyders) [reply src/fondant/component/data_io.py:231]: In our setup there should only be one client, and `get_client` retrieves the client connected to the current worker. So this should provide the correct one.

File: src/fondant/component/data_io.py [hunk 2]
```diff
@@ -204,7 +207,34 @@ def _write_dataframe(self, dataframe: dd.DataFrame) -> dd.core.Scalar:
...
+        futures = client.compute(to_parquet_tasks)
+
+        # As each future completes, release it so the memory can be reclaimed
+        for future in as_completed(futures):
+            future.result()
+            future.release()
 
     @staticmethod
     def _create_write_task(
```
[the remark this answers is not in the document]
Comment (RobbeSneyders) [reply src/fondant/component/data_io.py]: Good catch!

**Pull request merged**